lib/class-wp-rest-plugins-controller.php

```diff
@@ -0,0 +1,829 @@
+<?php
+/**
+ * Start: Include for phase 2
+ * Block Directory REST API: WP_REST_Plugins_Controller class
+ *
+ * @since   6.5.0
+ * @package gutenberg
+ */
+
+/**
+ * Core class to access plugins via the REST API.
+ *
+ * @since 5.5.0
+ *
+ * @see   WP_REST_Controller
+ */
+class WP_REST_Plugins_Controller extends WP_REST_Controller {
+
+	const PATTERN = '[^.\/]+(?:\/[^.\/]+)?';
+
+	/**
+	 * Plugins controller constructor.
+	 *
+	 * @since 5.5.0
+	 */
+	public function __construct() {
+		$this->namespace = '__experimental';
+		$this->rest_base = 'plugins';
+	}
+
+	/**
+	 * Registers the routes for the plugins controller.
+	 *
+	 * @since 5.5.0
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_item' ),
+					'permission_callback' => array( $this, 'create_item_permissions_check' ),
+					'args'                => array(
+						'slug'   => array(
+							'type'        => 'string',
+							'required'    => true,
+							'description' => __( 'WordPress.org plugin directory slug.', 'gutenberg' ),
+						),
+						'status' => array(
+							'description' => __( 'The plugin activation status.', 'gutenberg' ),
+							'type'        => 'string',
+							'enum'        => is_multisite() ? array( 'inactive', 'active', 'network-active' ) : array( 'inactive', 'active' ),
```

Comment:
Should there be a `network-inactive`?

Reply:
I think that just complicates the API surface. How would `network-inactive` be different from `inactive`?